libcxx/include/__compare/ordering.h

```diff
@@ -30,14 +30,17 @@ class partial_ordering;
 class weak_ordering;
 class strong_ordering;
 
-template <class _Tp, class... _Args>
-inline constexpr bool __one_of_v = (is_same_v<_Tp, _Args> || ...);
-
 struct _CmpUnspecifiedParam {
-  _LIBCPP_HIDE_FROM_ABI constexpr _CmpUnspecifiedParam(int _CmpUnspecifiedParam::*) noexcept {}
-
-  template <class _Tp, class = enable_if_t<!__one_of_v<_Tp, int, partial_ordering, weak_ordering, strong_ordering>>>
-  _CmpUnspecifiedParam(_Tp) = delete;
+  template <class _Tp, class = __enable_if_t<is_same_v<_Tp, int>>>
+  _LIBCPP_HIDE_FROM_ABI consteval _CmpUnspecifiedParam(_Tp __zero) noexcept {
```

Comment:
There's also a question of whether we want to make the use of other-than-literal-zero ill-formed or SFINAE-friendly. I think we're allowed to do whatever we want here since the Standard only says it's UB, but I'd welcome other people's thoughts on this.

Reply:
Ill-formed make sense to me, 
The scenario would be the presence of a `<=>` operator overload that returns something that is not a standard comparison category, but compares to `0` ? Seems unlikely